portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,168 @@
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.*;  
+import com.google.gson.Gson;
+import java.util.List;
+import java.util.Map;
+import java.util.Date;
+import java.text.DateFormat;  
+import java.text.SimpleDateFormat;  
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * When the user submits the form, Blobstore processes the file upload and then forwards the request
+ * to this servlet. This servlet can then process the request using the file URL we get from
+ * Blobstore.
+ */
+@WebServlet("/formhandler")
+public class FormHandlerServlet extends HttpServlet {
+    //milleseconds since epoc
+    //timestamps System.currentTimemillis
+    //javasortby timestamp
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  
+    Query query = new Query("CommentInfo").addSort("timestampL", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    
+    List<CommentInfo> CommentInfoList = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+        long timestampL = (long) entity.getProperty("timestampL");
+        String timestampS = (String) entity.getProperty("timestampS");
+        String comment = (String) entity.getProperty("comment");
+        String imageurl = (String) entity.getProperty("imgurl");
+       
+        CommentInfo commentinfo = new CommentInfo(timestampL, timestampS, comment, imageurl);
+
+        CommentInfoList.add(commentinfo);
+    }
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(CommentInfoList));//too json list ds containing class
+  }
+
+    
+    
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    // Get the message entered by the user.
+    String comment = request.getParameter("comment");
+    // Get the URL of the image that the user uploaded to Blobstore.
+    String imageurl = getUploadedFileUrl(request, "image");
```

Comment:
The walkthrough example only used blobstore to immediately show the image in response HTML. That example doesn't have a good story for deleting blobs when they are not needed. To be able to delete a blob you need to store its BlobKey in datastore, not its serving url. This way when you decide to delete the blob you can pass its blob key to BlobStore.delete() method. 

I'm not asking you to adjust this PR, but when working with blob store keep in mind how unused blobs will be deleted.